sql/handler.cc

```diff
@@ -7743,6 +7814,9 @@ int handler::ha_write_row(uchar *buf)
   if (unlikely(error= binlog_log_row(table, 0, buf, log_func)))
     DBUG_RETURN(error); /* purecov: inspected */
 
+  if (!is_fake_change_enabled(ha_thd()))
```

Comment:
The use case is I/O bound, and we have to consider the needs of 99.95% users who don't use it :)